gabbi/suitemaker.py

```diff
@@ -219,10 +219,10 @@ def test_suite_from_dict(loader, test_base_name, suite_dict, test_directory,
     default_test_dict = copy.deepcopy(case.HTTPTestCase.base_test)
     for handler in handlers:
         default_test_dict.update(handler.test_base)
-        if handler.response_handler:
-            response_handlers.append(handler.response_handler)
         if handler.content_handler:
             content_handlers.append(handler.content_handler)
+        elif handler.response_handler:
+            response_handlers.append(handler.response_handler)
```

Comment:
again this is probably not the right fix, as when a handler registers it can declare itself as both a content handler and a response handler and they are not mutually exclusive. For some things (notably the JSONHandler) it is both.

Reply:
I added a class attribute so a handler can declare itself as just a content handler.

Reply:
I noticed this change hasn't been reverted yet. Looks like there is still some more work to do on this part.

Reply:
I've got a branch in progress that tries to address that in a slightly different way. We'll want to compare notes on that and decide which is best. I'll push it up shortly.

Reply:
ok I'll wait on that then. I found there was a typo in jsonhandler.py too that this was covering up.
-    add_reponse_handling = False
+    add_response_handling = False